### PR TITLE
Fix: CollisionObject shape owner indexing is inconsistent

### LINKED ISSUE #11525: CollisionObject shape owner indexing is inconsistent
3.0 master

Shape owners added via `create_shape_owner()` start at id 1, but shapes added to those owners with `shape_owner_add_shape()` are indexed starting at 0.

I feel like shape owner ids should start at 0 as well. Am I wrong/missing something?

    create_shape_owner(self)
	print(get_shape_owners())  # prints [1]
	shape_owner_add_shape(1, CircleShape2D.new())
	print(shape_owner_get_shape(1, 0))  # the new circle shape

### PR DESCRIPTION
**Pull request by damarindra**
Fixes #11525

### CODE REVIEW

File: scene/2d/collision_object_2d.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ uint32_t CollisionObject2D::create_shape_owner(Object *p_owner) {
 	uint32_t id;
 
 	if (shapes.size() == 0) {
-		id = 1;
+		id = 0;
 	} else {
 		id = shapes.back()->key() + 1;
```
Comment (Noshyaar) [inline scene/2d/collision_object_2d.cpp]: `+ 1` here should be removed?

File: scene/3d/collision_object.cpp [hunk 2]
```diff
@@ -161,7 +161,7 @@ uint32_t CollisionObject::create_shape_owner(Object *p_owner) {
 	uint32_t id;
 
 	if (shapes.size() == 0) {
-		id = 1;
+		id = 0;
 	} else {
 		id = shapes.back()->key() + 1;
```
Comment (Noshyaar) [inline scene/3d/collision_object.cpp]: Also `+ 1` here

**Comment by Noshyaar** [comment]
Please use `git commit --amend` to push further changes. Thanks for contributing! 
([PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request))

File: scene/2d/collision_object_2d.cpp (hunk 1 above)
Comment (damarindra) [reply scene/2d/collision_object_2d.cpp]: I don't think so, isn't removing the `+1` will make the id same as the last one? I've compiled the source and test the `create_shape_owner` 2 times, the result is right `[0, 1]`. Or am I missing something?

Comment (Noshyaar) [reply scene/2d/collision_object_2d.cpp]: Oh, okay. I misunderstood the logic. 😝

**Review by ghost — APPROVED** [review]

**Pull request merged**